### ISSUE TITLE
Table: Checkbox click is showing incorrect selction

### ISSUE THREAD

**Issue by AnkitaBagale** [razorpay/blade#2271]
onSelectionChange is called with incorrect values on page change.

https://stackblitz.com/edit/mytgza-efebsi?file=App.tsx

https://github.com/razorpay/blade/assets/69760792/cdd9ac97-e456-4902-963a-1ad57a2267d8

**Comment by snitin315** [comment]
@AnkitaBagale It's working as expected in the [Basic Table story](http://localhost:9009/?path=/story/components-table--table&args=selectionType:multiple). I believe there is some other issue.

Also `x items selected` is the default selected title so you can just remove the `selectedItem` logic in your example.

**Comment by AnkitaBagale** [comment]
@snitin315 
If I remove selectedItem logic, I won't be able to access selectedItems ids.

**Comment by anuraghazra** [comment]
This is fixed in #2282 

For server side paginated data or partial data it is now recommended to use the `selectedIds` value exposed from `onSelectionChange` function. 

```ts
<Table onSelectionChange={({ selectedIds }) => {
  setSelectedIds(selectedIds)
}} />
```
